src/data/sonata_data.cpp

```diff
@@ -357,6 +357,12 @@ void SonataData::write_data(const std::vector<float>& buffered_data, uint32_t st
 
 void SonataData::flush() {
     write_data(report_buffer_, current_step_);
+    if (getenv("LIBSONATAREPORT_ENABLE_FLUSH") != nullptr) {
```

Comment:
Could this be read in the constructor and then reused as a member variable? Would maybe reduce the amount of string lookups :)